### PR TITLE
Payment logic framework

### PR DESCRIPTION
**Pull request by ryscher**
Adding the framework for payment logic. Adding an explicit check for institutional coverage of the Data Processing Charge.

To test:
- In a tenant file, set `covers_dpc: true`. 
- When a submission is made with this tenant active, the "Payment" paragraph will not appear on the "Review and Submit" page.
- When an item is published (or embargoed) with this tenant active, a Stripe invoice will not be generated.

Journal payments and fee waivers are not covered in this PR; they default to `false`.

### CODE REVIEW

**Comment by sfisher** [comment]
I saw what you added for the config repo, somehow I missed it but accepted it.

I think maybe some super simple test for user_must_pay would be nice.  I know it's a super simple method, yet in the future might expand more so would be nice to have at least a basic placeholder test for it.

**Comment by ryscher** [comment]
Ok @sfisher, this is now running tests. 

Note that I bumped the version of HTTParty, which required an update to `dryad` (https://github.com/CDL-Dryad/dryad/pull/59).

File: stash_engine/spec/config/app_config.yml [hunk 1]
```diff
@@ -5,6 +5,8 @@ test:
   repository: Stash::MockRepository
   submission_bc_emails: ["alan.smithee@example.edu"]
   contact_email: ["contact1@example.edu", "contact2@example.edu"]
+  old_dryad_access_token: "this-is-not-a-real-token"
+  old_dryad_url: http://ryan-vm.datadryad.org
```
Comment (briri) [inline stash_engine/spec/config/app_config.yml]: probably not a big deal but do you want your VM as part of the example config?

Comment (ryscher) [reply stash_engine/spec/config/app_config.yml]: fixed

**Comment by sfisher** [comment]
Thanks Ryan.  Merging.

**Pull request merged**